src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/slurm/SlurmJobManager.groovy

```diff
@@ -1,111 +1,402 @@
+
+    @Override
+    protected int getColumnOfJobState() {
+        return 1
+    }
+
+    @Override
+    protected Map<BEJobID, JobState> queryJobStates(List<BEJobID> jobIDs,
+                                                    Duration timeout = Duration.ZERO) {
+        StringBuilder queryCommand = new StringBuilder(getQueryJobStatesCommand())
+        if (jobIDs && !jobIDs.empty) {
+            queryCommand << " -j " << jobIDs.collect { it }.join(",")
+        }
+
+        if (isTrackingOfUserJobsEnabled)
+            queryCommand << " -u $userIDForQueries "
+
+        ExecutionResult er = executionService.execute(queryCommand.toString())
+        List<String> resultLines = er.stdout
+
+        Map<BEJobID, JobState> result = [:]
+
+        if (!er.successful) {
+            throw new BEException("Execution failed. ${er.toStatusLine()}")
+        } else {
+            if (resultLines.size() > 0) {
+                for (String line : resultLines) {
+                    line = line.trim()
+                    if (line.length() == 0) continue
+                    if (!RoddyConversionHelperMethods.isInteger(line.substring(0, 1)))
+                        continue //Filter out lines which have been missed which do not start with a number.
+
+                    String[] split = line.split("\\|")
+                    final int ID = getColumnOfJobID()
+                    final int JOBSTATE = getColumnOfJobState()
+
+                    BEJobID jobID = new BEJobID(split[ID])
+                    JobState js = parseJobState(split[JOBSTATE])
+
+                    result.put(jobID, js)
+                }
+            }
+        }
+        return result
+    }
+
+    @Override
+    protected JobState parseJobState(String stateString) {
+        JobState js = JobState.UNKNOWN
+
+        if (stateString == "RUNNING")
+            js = JobState.RUNNING
+        if (stateString == "SUSPENDED")
+            js = JobState.SUSPENDED
+        if (stateString == "PENDING")
+            js = JobState.HOLD
+        if (stateString == "CANCELLED+")
+            js = JobState.ABORTED
+        if (stateString == "COMPLETED")
+            js = JobState.COMPLETED_SUCCESSFUL
+        if (stateString == "FAILED")
+            js = JobState.FAILED
+
+        return js
+    }
+
+    @Override
+    Map<BEJobID, GenericJobInfo> queryExtendedJobStateById(List<BEJobID> jobIds,
+                                                           Duration timeout = Duration.ZERO) {
+        Map<BEJobID, GenericJobInfo> queriedExtendedStates = [:]
+        for (int i = 0; i < jobIds.size(); i++) {
+            ExecutionResult er = executionService.execute(getExtendedQueryJobStatesCommand() + " " + jobIds[i] + " -o")
+
+            if (er != null && er.successful) {
+                queriedExtendedStates = this.processExtendedOutput(er.stdout.join("\n"), queriedExtendedStates)
+            } else {
+                er = executionService.execute("sacct -j " + jobIds[i] + " -o --json")

```

Comment:
I noticed that also in WESkit (from a different programmer though :) ). You use `sacct`, which should report accounting information. When looking at the [Rosetta Stone Document on the SLURM website](https://slurm.schedmd.com/rosetta.pdf) the matching command for `bjobs` from LSF for retrieving job information is `squeue`. I **guess** that accounting information is not guaranteed to be always the same as job status, and I think it would be safer to rely on `squeue` instead.

But then this is also some kind of second level query, in case the first fails, and the first uses `scontrol` (which I don't know). Would you comment in the code why this logic was chosen? Under what conditions will the first command fail and make the second necessary? Could it be that with `squeue` this problem would not occur?

Reply:
I use `squeue` to retrieve the job status but not for extended output. The reason is, that in `scontrol` the information about the job is still some minutes (I think 5 min) available after the job has finished. Yes, I agree with `sacct` and it's only used in case some minutes is not enough, but I'm not sure if this scenario is very likely.